lib/service/service.go

```diff
@@ -3712,7 +3712,7 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 	// TCP applications.
 	if alpnRouter != nil {
 		alpnRouter.Add(alpnproxy.HandlerDecs{
-			MatchFunc: alpnproxy.MatchByProtocol(alpncommon.ProtocolTCP),
+			MatchFunc: alpnproxy.MatchByProtocolWithPing(alpncommon.ProtocolTCP),
```

Comment:
nit: can we make the `MatchByProtocol` aware about ping protocols support and call the WithPingProtocols internal. 

This will simplify the complexity and reduce the number of places that needs to be extended in case of adding a new protocol with ping support ?